frontend/packages/ceph-storage-plugin/src/types.ts

```diff
@@ -248,3 +248,41 @@ export type NavUtils = {
   getIndex: (searchSpace: any, search: string, offset?: number) => number;
   getAnchor: (step: number, mode: number) => string;
 };
+export type StoreAction =
+  | { type: 'setSecretName'; value: string }
+  | { type: 'setSecretKey'; value: string }
+  | { type: 'setAccessKey'; value: string }
+  | { type: 'setRegion'; value: string }
+  | { type: 'setTarget'; value: string }
+  | { type: 'setEndpoint'; value: string };
```

Comment:
we create `reducer.ts` file to store actions and state related to a store.
See `/ocs-install/internal-mode/`